chain/chunks/src/lib.rs

```diff
@@ -178,53 +206,128 @@ impl SealsManager {
         parent_hash: &CryptoHash,
         height: BlockHeight,
         shard_id: ShardId,
-    ) -> Result<&mut Seal, near_chain::Error> {
-        Ok(self.seals.entry(chunk_hash.clone()).or_insert({
-            let chunk_producer = self.runtime_adapter.get_chunk_producer(
-                &self.runtime_adapter.get_epoch_id_from_prev_block(parent_hash)?,
-                height,
-                shard_id,
-            )?;
-            let mut candidates = vec![];
-            for part_ord in 0..self.runtime_adapter.num_total_parts() {
-                let part_ord = part_ord as u64;
-                let part_owner = self.runtime_adapter.get_part_owner(parent_hash, part_ord)?;
-                if part_owner == chunk_producer || Some(part_owner) == self.me {
-                    continue;
-                }
-                candidates.push(part_ord);
+    ) -> Result<Seal, near_chain::Error> {
+        match self.past_seals.get(&height) {
+            Some(hashes) if hashes.contains(chunk_hash) => Ok(Seal::Past),
+
+            // None | Some(hashes) if !hashes.contains(chunk_hash)
```

Comment:
Should this be removed? Or is it a comment?

Reply:
It explains what is `_`. I agree with @birchmd to keep this comment.